examples/demo-rollup/src/rollup.rs

```diff
@@ -0,0 +1,94 @@
+use anyhow::Context;
+use celestia::verifier::RollupParams;
+use celestia::CelestiaService;
+use demo_stf::app::{App, DefaultContext};
+use demo_stf::runtime::get_rpc_methods;
+use risc0_adapter::host::Risc0Verifier;
+use sov_db::ledger_db::LedgerDB;
+use sov_rollup_interface::services::da::DaService;
+use sov_rollup_interface::zk::Zkvm;
+use sov_state::storage::Storage;
+use sov_stf_runner::{from_toml_path, RollupConfig, RunnerConfig, StateTransitionRunner};
+use tracing::{debug, Level};
+
+#[cfg(feature = "experimental")]
+use crate::register_rpc::register_ethereum;
+use crate::register_rpc::{register_ledger, register_sequencer};
+use crate::{get_genesis_config, initialize_ledger, ROLLUP_NAMESPACE};
+
+/// Dependencies needed to run the rollup.
+pub struct Rollup<Vm: Zkvm, DA: DaService> {
+    app: App<Vm, DA::Spec>,
+    da_service: DA,
+    ledger_db: LedgerDB,
+    runner_config: RunnerConfig,
+}
+
+/// Creates celestia based rollup.
+pub async fn new_rollup_with_celestia_da(
+    rollup_config_path: &str,
+) -> Result<Rollup<Risc0Verifier, CelestiaService>, anyhow::Error> {
+    debug!("Starting demo rollup with config {}", rollup_config_path);
+    let rollup_config: RollupConfig<celestia::DaServiceConfig> =
+        from_toml_path(rollup_config_path).context("Failed to read rollup configuration")?;
+
+    // Initializing logging
+    let subscriber = tracing_subscriber::fmt()
+        .with_max_level(Level::INFO)
+        .finish();
+    tracing::subscriber::set_global_default(subscriber)
+        .map_err(|_err| eprintln!("Unable to set global default subscriber"))
+        .expect("Cannot fail to set subscriber");
```

Comment:
This should probably be inside a `init_logging` function or similar, and called directly from `main` since it modifies global logging settings.